usd_rate/models/account_analytic_account.py

```diff
@@ -0,0 +1,66 @@
+from collections import defaultdict
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = "account.analytic.account"
+
+    usd_currency_id = fields.Many2one(
+        string="Usd currency", comodel_name="res.currency", default=lambda self: self.env.ref("base.USD").id
+    )
+    balance_usd = fields.Monetary(
+        compute="_compute_debit_credit_balance_usd", string="Balance (USD)", currency_field="usd_currency_id"
+    )
+    debit_usd = fields.Monetary(
+        compute="_compute_debit_credit_balance_usd", string="Debit (USD)", currency_field="usd_currency_id"
+    )
+    credit_usd = fields.Monetary(
+        compute="_compute_debit_credit_balance_usd", string="Credit (USD)", currency_field="usd_currency_id"
+    )
+
+    # Reference: analytic/models/analytic_account.py
+    # _compute_debit_credit_balance
+    @api.depends("line_ids.amount")
+    def _compute_debit_credit_balance_usd(self):
+        Curr = self.env["res.currency"]
+        analytic_line_obj = self.env["account.analytic.line"]
+        domain = [("account_id", "in", self.ids), ("company_id", "in", [False] + self.env.companies.ids)]
+        if self._context.get("from_date", False):
+            domain.append(("date", ">=", self._context["from_date"]))
+        if self._context.get("to_date", False):
+            domain.append(("date", "<=", self._context["to_date"]))
+        if self._context.get("tag_ids"):
+            tag_domain = expression.OR([[("tag_ids", "in", [tag])] for tag in self._context["tag_ids"]])
+            domain = expression.AND([domain, tag_domain])
+
+        user_currency = self.env.company.currency_id
+        credit_groups = analytic_line_obj.read_group(
+            domain=domain + [("amount_usd", ">=", 0.0)],
+            fields=["account_id", "currency_id", "amount_usd"],
+            groupby=["account_id", "currency_id"],
+            lazy=False,
+        )
+        data_credit = defaultdict(float)
+        for l in credit_groups:
+            data_credit[l["account_id"][0]] += Curr.browse(l["currency_id"][0])._convert(
+                l["amount_usd"], user_currency, self.env.company, fields.Date.today()
```

Comment:
Should be like this right as if the record currency is different than the user currency it will convert it. But we want it in usd always.
```suggestion
            data_credit[l["account_id"][0]] += l["amount_usd"]
```

Reply:
@tong-odoo

Reply:
Same below

Reply:
I agree this change. Add up all the amount_usd is enough as long as we can ensure that amount_usd in account.analytic.line is usd currency then it is okay.